core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -149,7 +149,14 @@ private[spark] class Executor(
       tr.kill(interruptThread)
     }
   }
-
+  def killAllTasks (interruptThread: Boolean) : Unit = {
```

Comment:
1. Done 
2. We are targeting a use case where in executors are launched inside another running process and when an executor goes down, it does not take the parent parent process down. In such cases, executors should kill the running tasks when they go down.  Given that, the runningtasks a private val, we need a method that can be called from the executor backend.